create-tiny-url.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+type UrlObject struct {
+	Url string
+}
+
+type UrlStoredObject struct {
+	Url  string
+	Tiny string
+}
+
+type UrlStoredObjectSet []UrlStoredObject
+
+var filename string = "./url_mappings.json"
+
+func generateTinyUrl(w http.ResponseWriter, r *http.Request) {
+	var urlObject UrlObject
+	reqBody, _ := ioutil.ReadAll(r.Body)
+
+	err := json.Unmarshal(reqBody, &urlObject)
+
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	tinyUrl := createHashForUrl(urlObject.Url)
+
+	fmt.Fprint(w, "https://t.url/"+tinyUrl)
+}
+
+func createHashForUrl(url string) string {
+	urlStoredObject := ReadFromFIle(filename)
+
+	// check if current url exists
+	for _, urlObj := range urlStoredObject {
+		if urlObj.Url == url {
+			return urlObj.Tiny
+		}
+	}
+
+	// create new mapping
+	// store in file and return
+	urlHashBytes := md5.Sum([]byte(url))
+	urlHashString := hex.EncodeToString(urlHashBytes[:3])
+
+	modData := urlStoredObject.toString() + `{"url":"` + url + `","tiny":"https://t.url/` + urlHashString + `"}]`
```

Comment:
modData - can be an object with json.

```go
// Url ...
type Url struct {
    Url string `json:"url"
    TinyUrl string `json:"tiny"
}
```

```go
url := tiny.New("your long url") // this method return a URL struct
jsonData, err := json.Marsall(url)
if err != nil {
    return err
}
```

2) It's better to divide methods:
a) create an URL - `url := tiny.New("long url) Url` /// models
b) save it to file - `repo.Save(url) error` // storage
c) add a service - where you can use repo and url.